tests/integration/buildcmd/test_build_cmd.py

```diff
@@ -1560,6 +1560,41 @@ def test_cache_build(self, use_container, code_uri, function1_handler, function2
                 expected_messages, command_result, self._make_parameter_override_arg(overrides)
             )
 
+    def test_no_cached_override_build(self):
+        overrides = {
+            "FunctionCodeUri": "Python",
+            "Function1Handler": "main.first_function_handler",
+            "Function2Handler": "main.second_function_handler",
+            "FunctionRuntime": "python3.8",
+        }
+        config_file_testdata = Path(__file__).resolve().parents[2].joinpath("integration", "testdata")
+        config_file_buildcmd = config_file_testdata.joinpath("buildcmd")
+        config_file = str(config_file_buildcmd.joinpath("samconfig_no_cached.toml"))
+        cmdlist = self.get_command_list(parameter_overrides=overrides, cached=True)
+        command_result = run_command(cmdlist, cwd=self.working_dir)
+        self.assertTrue(self.default_build_dir.exists())
+        self.assertIn("samconfig_no_cached.toml", os.listdir(str(config_file_buildcmd)))
```

Comment:
nit: These two assertions probably aren't necessary since we expect the `build_dir` to exist in the context of this test and we manually created the config file.

Reply:
Sure, I will remove them